provider/resource_keycloak_required_action.go

```diff
@@ -0,0 +1,129 @@
+package provider
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/mrparkers/terraform-provider-keycloak/keycloak"
+)
+
+func resourceKeycloakRequiredAction() *schema.Resource {
+
+	return &schema.Resource{
+		Create: resourceKeycloakRequiredActionsCreate,
+		Read:   resourceKeycloakRequiredActionsRead,
+		Delete: resourceKeycloakRequiredActionsDelete,
+		Update: resourceKeycloakRequiredActionsUpdate,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"realm_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"alias": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"default_action": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Optional: true,
```

Comment:
1. Keycloak does not have a problem with same priorities. It just uses priority to order them. It then falls back on alphabetical order.

keycloak Required Action comparator code (/home/tom/projects/keycloak/source/keycloak/server-spi/src/main/java/org/keycloak/models/RequiredActionProviderModel.java:34):

`
 @Override
        public int compare(RequiredActionProviderModel o1, RequiredActionProviderModel o2) {

            return Comparator
                    .comparingInt(RequiredActionProviderModel::getPriority)
                    .thenComparing(RequiredActionProviderModel::getName, Comparator.nullsFirst(String.CASE_INSENSITIVE_ORDER))
                    .compare(o1, o2);
        }
`

2. If you not fill in a priority, terraform fills in 0, which imho is also not a problem. 


Personally, I do not think we need to put here extra logic. it has sensible fallbacks if you not fill it in or fill in the same priority with different required actions.....

Reply:
Makes sense to make, thanks for digging into Keycloak's source code for this!

I still think it makes sense to add `Computed` to this attribute, because it sounds like it will be set by Keycloak if it isn't provided.